app/src/main/java/com/truongdc21/quickquotes/ui/adapter/QuotesAdapter.kt

```diff
@@ -37,6 +52,178 @@ class QuotesAdapter(
+        imgFavoriteAuthor: ImageView,
+        tvFavoriteAuthor: TextView
+    ){
+        CoroutineScope(Dispatchers.Default).launch {
+            launch {
+                var isCheck = true
+                for (i in mListAuthorLocal) {
+                    if (itemQuotes.Author == i.mAuthor) {
+                        withContext(Dispatchers.Main) {
+                            imgFavoriteAuthor.setBackgroundResource(R.drawable.ic_favorite2)
+                            tvFavoriteAuthor.text = mContext.resources.getString(R.string.remove_author_from_favorite)
+                        }
+                        isCheck = false
+                        break
+                        this.cancel()
+                    }
+                }
+                if (isCheck) {
+                    withContext(Dispatchers.Main) {
+                        imgFavoriteAuthor.setBackgroundResource(R.drawable.ic_favorite)
+                        tvFavoriteAuthor.text = mContext.resources.getString(R.string.add_author_favorite)
+                    }
+                }
+            }
+        }
+    }
+
+    private fun showButtonSheetDialog(
+        binding: LayoutItemQuotesBinding,
+        itemQuotes: Quotes,
+        position: Int
+    ){
+        val view = LayoutInflater.from(mContext).inflate(R.layout.layout_bottom_sheet_dialog , null)
+        val bottomSheetDialog = BottomSheetDialog(mContext, R.style.AppBottomSheetDialogTheme)
+        bottomSheetDialog.setContentView(view)
+        bottomSheetDialog.show()
+        val viewRemove = view.findViewById<LinearLayout>(R.id.viewRemove)
+        val viewShared = view.findViewById<LinearLayout>(R.id.viewShared)
+        val viewSaveAuthor = view.findViewById<LinearLayout>(R.id.viewSaveAuthor)
+        val viewSaveTag = view.findViewById<LinearLayout>(R.id.viewSaveTag)
+        val viewTranslate = view.findViewById<LinearLayout>(R.id.viewTranslate)
+        val imgFavoriteAuthor = view.findViewById<ImageView>(R.id.imgItemFavoriteAuthor)
+        val imgFavoriteTag = view.findViewById<ImageView>(R.id.imgItemFavoriteTag)
+        val tvFavoriteAuthor = view.findViewById<TextView>(R.id.tvItemFavoriteAuthor)
+        val tvFavoriteTag = view.findViewById<TextView>(R.id.tvItemFavoriteTag)
+        val imgHideBottomSheet = view.findViewById<ImageView>(R.id.imgHideBottomSheet)
+        val tvTranslate = view.findViewById<TextView>(R.id.tvTranslate)
+
+        checkTranslate(position, tvTranslate)
+        checkFavoriteAuthorBottomSheet(itemQuotes, imgFavoriteAuthor,tvFavoriteAuthor)
+        checkFavoriteTagBottomSheet(itemQuotes, imgFavoriteTag, tvFavoriteTag)
+
+        imgHideBottomSheet.setOnClickListener {
+            it.setAnimationClick()
+            bottomSheetDialog.dismiss()
+        }
+
+        viewRemove.setOnClickListener {
+            it.setAnimationClick()
+            mListQuotes.removeAt(position)
+            notifyItemRemoved(position)
+            bottomSheetDialog.dismiss()
```

Comment:
sử dụng nhiều onClicklistener thì ông viết mỗi action ra 1 hàm
```suggestion
           onClickViewRemove()
```